.github/workflows/lint.yml

```diff
@@ -2,26 +2,32 @@ name: Linting
 
 on:
   pull_request:
-  push:
-    branches:
-      - master
+
+permissions:
+  contents: read
+  pull-requests: read
+  checks: write
 
 jobs:
   golangci:
     name: lint
     runs-on: ubuntu-latest
     steps:
       - name: Check out code
-        uses: actions/checkout@v4
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          persist-credentials: false
 
       - name: golangci-lint
-        uses: reviewdog/action-golangci-lint@v2
+        uses: reviewdog/action-golangci-lint@dd3fda91790ca90e75049e5c767509dc0ec7d99b # v2
         with:
           reporter: github-pr-review
           fail_on_error: true
           filter_mode: nofilter
 
+      - name: Install editorconfig-checker
+        uses: editorconfig-checker/action-editorconfig-checker@d2ed4fd072ae6f887e9407c909af0f585d2ad9f4 # v2
+
       - name: eclint
-        uses: reviewdog/action-eclint@v1
-        with:
-          github_token: ${{ secrets.github_token }}
+        run: editorconfig-checker
```

Comment:
as this is a different action we will have to update the required checks before merging this PR